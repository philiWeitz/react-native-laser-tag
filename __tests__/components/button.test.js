/* global describe, it, expect */

import React from 'react';
import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import { TouchableOpacity } from 'react-native';
import Button from '../../src/components/button/index';


describe('component-button-tests', () => {

  it('button snapshot test', () => {
    const wrapper = shallow(
      <Button text="Button Test" onPress={() => {}} />,
    );
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });

  it('button press test', () => {
    let buttonPressed = false;
    const wrapper = shallow(
      <Button onPress={() => { buttonPressed = true; }} text="Button Test" />,
    );

    wrapper.find(TouchableOpacity).simulate('press');
    expect(buttonPressed).toBe(true);
  });

});
