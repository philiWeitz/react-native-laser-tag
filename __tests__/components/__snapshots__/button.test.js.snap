// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component-button-tests button snapshot test 1`] = `
<Button
  borderRadius={5}
  buttonStyle={
    Object {
      "backgroundColor": "#008CBA",
      "flexDirection": "row",
    }
  }
  containerViewStyle={
    Array [
      Object {
        "marginLeft": 0,
        "marginRight": 0,
      },
      Object {},
    ]
  }
  icon={Object {}}
  iconRight={Object {}}
  onPress={[Function]}
  raised={true}
  textStyle={
    Object {
      "textAlign": "center",
    }
  }
  title="Button Test"
/>
`;
